report_qweb_table_pagebreak/models/ir_qweb.py

```diff
@@ -0,0 +1,104 @@
+from lxml import html
+
+from odoo import api, models
+
+from odoo.addons.base.models.qweb import QWeb
+
+
+class StyleParser:
+    """Class to manipulate and Serialize/DeSerialize style attribute on
+    html tag::
+
+        style = StyleParser(element.get("style"))
+        # set attribute page-break-inside
+        style.set("page-break-inside", "avoid")
+        # Serialize style to set style value on html element
+        element.set("style", str(style))
+    """
+
+    _styles: dict = {}
+
+    def __init__(self, style_content: str):
+        self._parse(style_content)
+
+    def _parse(self, style_content):
+        self._styles = {}
+        for style in style_content.split(";"):
+            data_style = style.split(":")
+            if len(data_style) != 2:
+                # ignore if not expected string
+                continue
+            key, value = data_style
+            self._styles[key.strip()] = value.strip()
+
+    def __str__(self):
+        return "; ".join([f"{k}: {v}" for k, v in self._styles.items()])
+
+    def set(self, key, value, default=None):
+        if value is None:
+            value = default
+        if value is None:
+            self._styles.pop(key, None)
+        else:
+            self._styles[key] = value
+
+
+class IrQWeb(models.AbstractModel, QWeb):
+    _inherit = "ir.qweb"
+    _description = "Qweb"
+
+    @api.model
+    def _render(self, id_or_xml_id, values=None, **options):
+        result = super()._render(id_or_xml_id, values=values, **options)
+
+        if b"data-page-break=" not in result:
+            return result
+
+        return self._split_tables(result)
+
+    @api.model
+    def _split_tables(self, html_content):
+        html_document = html.document_fromstring(html_content.decode("utf-8"))
+
+        for fragment in html_document:
+            table = None
+            for row in fragment.iterfind(".//tr[@data-page-break]"):
+                # while generating pdf this method is called twice so we prevent
+                # table to be split twice by removing data-page-break once
+                # table has been spited
+                pos = row.attrib.pop("data-page-break")
+                if pos not in ("before", "after", "avoid"):
+                    continue
+                table = next(row.iterancestors("table"))
+                tbody = next(row.iterancestors("tbody"))
+                newtable = html.Element("table", attrib=dict(table.attrib))
```

Comment:
`style` attribute and all other  `tbody`/`table` attributes are copied on new element while creating the element with the  `attrib=dict(tbody.attrib)`. There is a unit test on `table` element, I'll add it on tbody as well

Reply:
Sorry I was not clear enough, I was talking about CSS rules like
```
table {
  ...
}
```
that are applied to any `table`; on a second thought that also applies to all the attributes of the `table` node.

What I meant is: the user has added some style to the main table of the report, maybe it is not expected to see that same style applied to another table _inside_ the list of rows?
Maybe the new table should be as invisible as possible to mimic the visual appearance of normal rows?
I wouldn't know how to do that but I was just saying that it might be something to consider.

Reply:
yeah hard to find the best solution, I'll add a comment to warn developer that there are some magic that generate multiple tables instead of one.

And just realized that this code is also called while templates are shared between PDF report and portail views (likes for customer invoices)